lib/token.ts

```diff
@@ -724,14 +736,59 @@ function renewToken(sdk: OktaAuth, token: Token): Promise<Token> {
     userinfoUrl,
     issuer
   })
-  .then(function(res) {
-    // Multiple tokens may have come back. Return only the token which was requested.
-    var tokens = res.tokens;
-    return isIDToken(token) ? tokens.idToken : tokens.accessToken;
-  });
+    .then(function (res) {
+      // Multiple tokens may have come back. Return only the token which was requested.
+      var tokens = res.tokens;
+      return isIDToken(token) ? tokens.idToken : tokens.accessToken;
+    });
+}
+
+async function renewTokensWithRefresh(sdk: OktaAuth, options: TokenParams, refreshTokenObject: RefreshToken): Promise<any> {
```

Comment:
Looks like the return type is TokenResponse instead of Tokens, which means the tokenManager cannot get proper tokens after the renew process.